custom_components/mojelektro/sensor.py

```diff
@@ -81,20 +84,53 @@ def __init__(self, coordinator, entry_id, measurement_name, meter_id, hass):
         self.measurement_name = measurement_name
         self._last_known_state = None
         
+        self._attr_unit_of_measurement = self.get_unit_of_measurement(measurement_name)
+        self._attr_native_unit_of_measurement = self.to_native_unit_of_measurement(self._attr_unit_of_measurement)
+
         if self.measurement_name.startswith('casovni_blok'):
             # For casovni_blok sensors
-            self._attr_native_unit_of_measurement = UnitOfPower.KILO_WATT
-            self._attr_unit_of_measurement = "kW"  # Direct string to avoid any confusion
             self._attr_device_class = SensorDeviceClass.POWER  # Use POWER device class for power sensors
             self._attr_icon = "mdi:flash"
         else:
             # For other sensors
-            self._attr_native_unit_of_measurement = UnitOfEnergy.KILO_WATT_HOUR
-            self._attr_unit_of_measurement = "kWh"  # Direct string to avoid any confusion
             self._attr_device_class = SensorDeviceClass.ENERGY
             self._attr_state_class = SensorStateClass.TOTAL_INCREASING
             self._attr_icon = "mdi:transmission-tower"
 
+    def to_native_unit_of_measurement(self, unit_of_measurement):
+        if unit_of_measurement == "kW":
+            return UnitOfPower.KILO_WATT
+        elif unit_of_measurement == "kWh":
+            return UnitOfEnergy.KILO_WATT_HOUR
+        else:
+            _LOGGER.debug("missing conversion to native_unit_of_measurement for '%s'.", unit_of_measurement)
+            return None
+
+    def get_unit_of_measurement(self, measurement_name):
+        oznaka = self.get_oznaka(measurement_name)
+        
+        if oznaka == None:
+            if measurement_name.startswith('casovni_blok'):
+                return "kW"
+            else:
+                return "kWh"
+
+        for item in json.loads(READING_TYPE_ARRAY):
+            if item.get("oznaka") == oznaka:
+                return item.get("enota")
+
+    def get_oznaka(self, measurement_name):
```

Comment:
would be probably reasonable to separate this in future to new file with `find_tag` (and related) functions from `moj_elektro_api.py`

Reply:
Also, this can be surely done better it was just most straightforward....